keras/saving/hdf5_format.py

```diff
@@ -625,28 +645,53 @@ def save_weights_to_hdf5_group(f, layers):
   from keras import __version__ as keras_version  # pylint: disable=g-import-not-at-top
 
   save_attributes_to_hdf5_group(
-      f, 'layer_names', [layer.name.encode('utf8') for layer in layers])
+      f, 'layer_names', [layer.name.encode('utf8') for layer in model.layers])
   f.attrs['backend'] = backend.backend().encode('utf8')
   f.attrs['keras_version'] = str(keras_version).encode('utf8')
 
   # Sort model layers by layer name to ensure that group names are strictly
   # growing to avoid prefix issues.
-  for layer in sorted(layers, key=lambda x: x.name):
+  for layer in sorted(model.layers, key=lambda x: x.name):
     g = f.create_group(layer.name)
-    weights = _legacy_weights(layer)
-    weight_values = backend.batch_get_value(weights)
-    weight_names = [w.name.encode('utf8') for w in weights]
-    save_attributes_to_hdf5_group(g, 'weight_names', weight_names)
-    for name, val in zip(weight_names, weight_values):
-      param_dset = g.create_dataset(name, val.shape, dtype=val.dtype)
-      if not val.shape:
-        # scalar
-        param_dset[()] = val
-      else:
-        param_dset[:] = val
+    save_layer_weights_to_hdf5_group(g, layer)
+  save_layer_weights_to_hdf5_group(f, model)
```

Comment:
This doesn't seem right, because `model.weights` includes the weights of the sublayers.

You're going to want to do something like:

```python
for layer in sorted(model.layers, key=lambda x: x.name):
  g = f.create_group(layer.name)
  save_weights_subset_to_hdf5_group(g, _legacy_weights(layer))
g = f.create_group(top_level_weights)
save_weights_subset_to_hdf5_group(g, self._trainable_weights + self._non_trainable_weights)
```

Am I missing something?